lib/improver/tests/blending/weights/test_WeightsUtilities.py

```diff
@@ -278,18 +278,21 @@ def test_basic_weights(self):
         """Test building a cube with weights along the blending coordinate."""
 
         weights = np.array([0.4, 0.6])
-        blending_coord = 'time'
+        blending_coord = 'forecast_period'
 
         plugin = WeightsUtilities.build_weights_cube
         result = plugin(self.cube, weights, blending_coord)
-
+        expected_coord_names = ["time", "forecast_reference_time",
+                                "forecast_period"]
         self.assertIsInstance(result, iris.cube.Cube)
         self.assertEqual(result.name(), 'weights')
         self.assertFalse(result.attributes)
         self.assertArrayEqual(result.data, weights)
         self.assertEqual(result.coords(dim_coords=True)[0].name(),
-                         blending_coord)
+                         "time")
```

Comment:
Why these changes?

Reply:
This was trying to test the change I made to the build_weights_cube function to keep all the coordinates associated with the blending coord, not just the dim coord, so in this case we're keeping both the forecast_period and time coordinates, not just the forecast_period coordinate

Reply:
OK.  I think we do want to keep a test that blends across forecast period, as this is a very widely used case (and of course blends across an aux_coord).  Should this not also retain the time and forecast reference time coordinates (although only time shares a dimension with forecast period)?  I could see the case for a new test blending across time, but would rather not lose the existing one.

Reply:
I've now changed this to add a new unit test for this rather than editing the existing one. It is still doing blending over forecast period, and the cube doesn't quite look like the input cubes we would expect in blending in reality, but the test does test the new functionality. Hopefully this is ok.